azurelinuxagent/ga/agent_update_handler.py

```diff
@@ -34,107 +34,145 @@ class AgentUpdateHandlerUpdateState(object):
     This state will be persisted throughout the current service run.
     """
     def __init__(self):
-        self.last_attempted_requested_version_update_time = datetime.datetime.min
-        self.last_attempted_hotfix_update_time = datetime.datetime.min
-        self.last_attempted_normal_update_time = datetime.datetime.min
+        self.last_attempted_rsm_version_update_time = datetime.datetime.min
+        self.last_attempted_self_update_hotfix_time = datetime.datetime.min
+        self.last_attempted_self_update_regular_time = datetime.datetime.min
         self.last_attempted_manifest_download_time = datetime.datetime.min
+        self.last_attempted_update_error_msg = ""
+        self.last_attempted_update_version = FlexibleVersion("0.0.0.0")
 
 
 class AgentUpdateHandler(object):
-
+    """
+    This class handles two type of agent updates and chooses the appropriate updater based on the below conditions:
+        RSM update: This is the update requested by RSM. The contract between CRP and agent is we get following properties in the goal state:
+                    version: it will have what version to update
+                    isVersionFromRSM: True if the version is from RSM deployment.
+                    isVMEnabledForRSMUpgrades: True if the VM is enabled for RSM upgrades.
+                    if vm enabled for RSM upgrades, we use RSM update path. But if requested update is not by rsm deployment
+                    we ignore the update.
+                    This update is allowed once per (as specified in the conf.get_autoupdate_frequency())
```

Comment:
is this auto update frequency only used for RSM update? I see we have separate frequencies for hotfix and regular updates. If autoupdate_frequency is only used for RSM, could we update the name to differentiate it?

Reply:
same frequency used for manifest download in self-update, let me think better name